lib/src/main/proto/proto/faas/v1/faas.proto

```diff
@@ -0,0 +1,304 @@
+  // TODO: Remove in 1.0
+  map<string, string> headers_old = 3 [deprecated=true];
+  
+  // The old query params (preserving for backwards compatibility)
+  // TODO: Remove in 1.0
+  map<string, string> query_params_old = 4  [deprecated=true];
+
+  // HTTP request headers
+  map<string, HeaderValue> headers = 5;
+
+  // HTTP Query params
+  map<string, QueryValue> query_params = 6;
+
+  // HTTP Path parameters
+  map<string, string> path_params = 7;
+}
+
+message TopicTriggerContext {
+  // The topic the message was published for
+  string topic = 1;
+
+  // TODO: Add the event ID to the trigger context here got transactional outbox?
+}
+
+message BucketNotification {
+  string key = 1;
+  BucketNotificationType type = 2;
+}
+
+message NotificationTriggerContext {
+  string source = 1;
+  
+  // Now we don't need to specify a type enum, the specific contents/type comes in a single field of the context.
+  oneof notification {
+     BucketNotification bucket = 10;
+  }
+}
+
+message WebsocketTriggerContext {
+  // The nitric name of the socket that this worker listens on
+  string socket = 1;
+
+  // The type of websocket event
+  WebsocketEvent event = 2;
+
+  // The connection this trigger came from
+  string connectionId = 3;
+
+  // The query params available in the connection request
+  map<string, QueryValue> query_params = 6;
+}
+
+// The worker has successfully processed a trigger
+message TriggerResponse {
+  // The data returned in the response
+  bytes data = 1;
+
+  // The context of the request response
+  // Typically this will be one to one with the Trigger Context
+  // i.e. if you receive http context you may return http context
+  oneof context {
+    // response to a http request
+    HttpResponseContext http = 10;
+    // response to a topic trigger
+    TopicResponseContext topic = 11;
+    // response to a notification trigger
+    NotificationResponseContext notification = 12;
+
+    WebsocketResponseContext websocket = 13;
+  }
+}
+
+// Specific HttpResponse message
+// Note this does not have to be handled by the
+// User at all but they will have the option of control
+// If they choose...
+message HttpResponseContext {
+  // Old HTTP response headers (deprecated)
+  // TODO: Remove in 1.0
+  map<string, string> headers_old = 1 [deprecated=true];
+
+  // The HTTP status of the request
+  int32 status = 2;
+
+  // HTTP response headers
+  map<string, HeaderValue> headers = 3;
+}
+
+// Specific event response message
+// We do not accept responses for events
+// only whether or not they were successfully processed
+message TopicResponseContext {
+  // Success status of the handled event
+  bool success = 1;
+}
+
+message NotificationResponseContext {
+	bool success = 1;
+}
+
+message WebsocketResponseContext {
+  // There aren't really responses here, instead we need to provide a runtime
+  // API for responding back and emitting/broadcasting on the websocket from the server
+  bool success = 1;
+}
```

Comment:
can probably add thse to the .gitignore if we're downloading when we regenerate.

Reply:
removed